Hymns/Tags/TagSheetView.swift

```diff
@@ -0,0 +1,62 @@
+import SwiftUI
+import Resolver
+
+struct TagSheetView: View {
+
+    @ObservedObject private var viewModel: TagSheetViewModel
+    @Environment(\.presentationMode) var presentationMode
+    @State private var tagName = ""
+    @State private var tagColor = TagColor.none
+    var sheet: Binding<DisplayHymnSheet?>
+
+    init(viewModel: TagSheetViewModel, sheet: Binding<DisplayHymnSheet?>) {
+        self.viewModel = viewModel
+        self.sheet = sheet
+    }
+
+    var body: some View {
+        VStack(alignment: .leading) {
+            Text("Name your tag").font(.body).fontWeight(.bold)
+            TextField("Label it however you like", text: self.$tagName)
+            Divider()
+            ColorSelectorView(tagColor: self.$tagColor).padding(.vertical)
+            if !self.viewModel.tags.isEmpty {
+                Section {
+                    Text("Tags").font(.body).fontWeight(.bold)
+                    HStack {
+                        ForEach(self.viewModel.tags, id: \.self) { tag in
+                            Button(action: {
+                                //TODO FIX Delete is deleting all the tags because the button when clicked is clicking all of them.
```

Comment:
Why does this delete all the tags? Is it difficult to just fix that in this PR

Reply:
It's a wierd bug thing. It's a problem with the front end. I think it has to do with there being a bunch of buttons within a ForEach and how it SwiftUI reacts when one of them is clicked. Somehow they are all getting triggered sometimes. But also sometimes it doesn't happen. I need to figure out a solution for this. Because it is a significant problem.

Reply:
Nevermind. I found a solution. You just can't use button.                             //https://stackoverflow.com/questions/60960077/how-do-i-keep-multiple-button-actions-separate-in-swiftui-foreach-content Can't use a button here
I implemented the solution using onTapGesture. I left a comment in the code but maybe we don't need a code comment idk?